cmd/clusterctl/client/cluster/components.go

```diff
@@ -188,6 +188,9 @@ func (p *providerComponents) Delete(options DeleteOptions) error {
 
 		if util.IsClusterResource(obj.GetKind()) &&
 			!isNamespace && !isCRD && !isWebhook &&
+			// TODO(oscr) Delete the check below condition when the min version to upgrade from is CAPI v1.3
```

Comment:
I'm not opposed to this for now, but I think we should agree on some conventions for tracking tasks to happen sometime in the future vs assigning TODO to single contributors
cc @sbueringer

Reply:
My take:
* TODO(name) is okay but this can only ever be an additional data point that something has to get done
* TODO's overall are pretty much worthless if something actually should get done at a certain point (we have roughly 270 TODOs in our code (even without context.TODO()))
* So I use TODO's only as a hint that there's room for improvement if I see them when reading code


If things actually should be done:
* Create an issue
  * Ideally assign it to the corresponding milestone to actually get it tracked for a specific release
* We can come up with some conventions that we can grep for. I used something like `TODO(after Cluster API v1.4)`, `TODO(after Kubernetes v1.25)` relatively successful in the past. But we need those conventions and then we can add tasks to e.g. bump Cluster API / bump Kubernetes checklists.

Reply:
I think I agree
We can allow TODO as a hint that there's room for improvement; when there is something to be done in the future, there should be an issue and hopefully, we will catch it during backlog triages sessions